packages/plugin-ext/doc/how-to-add-new-plugin-namespace.md

```diff
@@ -90,23 +122,153 @@ function findPlugin(filePath: string): Plugin | undefined {
 }
 ```
 
-## Next you need to implement `createAPIFactory` factory function
+## Implement your API object
+
+We create a dedicated API object for each individual plugin as part of the module loading process.
+Each API object is returned as part of the module loading process if a script imports `@bar/foo` and should therefore match the API definition that we provided in the `*.d.ts` file.
+Multiple imports will not lead to the creation of multiple API objects as we cache it in our custom `overrideInternalLoad` function.
 
-Example:
+Example `foo-init.ts` (continued):
 
 ```typescript
-import * as fooApi from '@bar/foo';
 export function createAPIFactory(rpc: RPCProtocol): ApiFactory {
-    const fooBarImpl = new FooBarImpl(rpc);
-    return function (plugin: Plugin): typeof fooApi {
-        const FooBar: typeof fooApi.fooBar = {
-            getFoo(): fooApi.Foo{
-                return fooBarImpl.getFooImpl();
+    const fooExtImpl = new FooExtImpl(rpc);
+    return function (plugin: Plugin): typeof fooBarAPI {
+        const FooBar: typeof fooBarAPI.fooBar = {
+            getFoo(): Promise<fooBarAPI.Foo> {
+                return fooExtImpl.getFooImpl();
             }
         }
-        return <typeof fooApi>{
+        return <typeof fooBarAPI>{
             fooBar : FooBar
         };
     }
+}
+```
+
+In the example above the API object creates a local object that will fulfill the API contract.
+The implementation details are hidden by the object and it could be a local implementation that only lives inside the plugin host but it could also be an implementation that uses the `RPCProtocol` to communicate with the main application to retrieve additional information.
```

Comment:
"to retrieve additional information" may give the wrong idea to the reader. Often, an API triggers changes or drives functionality on the main side of things. Yes, sometimes it just gets information, but that's probably more the exception than the rule. The wording here should cover both cases.

Reply:
Yeah, I get that. Good catch! I adapted the sentence to cover more cases and give a better idea of what could be possible.